File: hector_moveit_navigation/src/navigator_client.cpp
#include <ros/ros.h>
#include <geometry_msgs/Pose.h>
#include <actionlib/client/simple_action_client.h>
#include <actionlib/client/terminal_state.h>
#include <hector_moveit_navigation/NavigationAction.h>

int main (int argc, char **argv)
{
  ros::init(argc, argv, "nagvigation_client");

  actionlib::SimpleActionClient<hector_moveit_navigation::NavigationAction> ac("hector_navigator", true);

  ROS_INFO("Waiting for action server to start.");
  ac.waitForServer();

  ROS_INFO("Action server started, sending goal.");

  hector_moveit_navigation::NavigationGoal goal;

  goal.goal_pose.position.x = 0;
  goal.goal_pose.position.y = 0;
  goal.goal_pose.position.z = 1;

  goal.goal_pose.orientation.x = 0;
  goal.goal_pose.orientation.y = 0;
  goal.goal_pose.orientation.z = 0;
  goal.goal_pose.orientation.w = 1;

  ROS_INFO("Send takeoff goal");
  ac.sendGoal(goal);

  bool finished_before_timeout = ac.waitForResult(ros::Duration(30.0));

  if(finished_before_timeout)
  {
  	actionlib::SimpleClientGoalState state = ac.getState();
  	ROS_INFO("Takeoff finished: %s", state.toString().c_str());
  }
  else
  	ROS_INFO("Takeoff did not finish before the time out.");

  goal.goal_pose.position.x = 2;
  goal.goal_pose.position.y = 2;
  goal.goal_pose.position.z = 2;

  ac.sendGoal(goal);

  finished_before_timeout = ac.waitForResult(ros::Duration(30.0));

  if(finished_before_timeout)
  {
  	actionlib::SimpleClientGoalState state = ac.getState();
  	ROS_INFO("Action finished: %s", state.toString().c_str());
  }
  else
  	ROS_INFO("Action did not finish before the time out.");

  return 0;
}

File: hector_moveit_navigation/src/navigator_client.py
#!/usr/bin/env python
import rospy
import actionlib
from geometry_msgs.msg import Pose

import hector_moveit_navigation.msg
import octomap_msgs.msg

def callback(data):
	rospy.loginfo("Got octomap marker array")

def navigator_client():
	rospy.init_node('navigator_client_py')
	rospy.loginfo("Initialized node")
	rospy.Subscriber("octomap_binary", octomap_msgs.msg.Octomap, callback)

	client = actionlib.SimpleActionClient('hector_navigator', hector_moveit_navigation.msg.NavigationAction)

	rospy.loginfo("Waiting for server")
	client.wait_for_server()

	goal = hector_moveit_navigation.msg.NavigationGoal();
	goal.goal_pose.position.x = 0;
	goal.goal_pose.position.y = 0;
	goal.goal_pose.position.z = 1;

	goal.goal_pose.orientation.x = 0;
	goal.goal_pose.orientation.y = 0;
	goal.goal_pose.orientation.z = 0;
	goal.goal_pose.orientation.w = 1;

	rate = rospy.Rate(2)
	while not rospy.is_shutdown():
		if goal.goal_pose.position.z == 5:
			goal.goal_pose.position.z = 1
		else:
			goal.goal_pose.position.z = goal.goal_pose.position.z + 1

		rospy.loginfo("Sending goal")
		client.send_goal(goal)

		rospy.loginfo("Waiting for result")
		client.wait_for_result()
		rate.sleep()
	

	

if __name__ == '__main__':
	try:
		navigator_client()
	except rospy.ROSInterruptException:
		pass